fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    async function getUser() {
+      try {
+        const url = `https://jsonplaceholder.typicode.com/users/${userId}`;
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error(`Response status: ${response.status}`);
+        }
+        const json = await response.json();
```

Comment:
You should use type assertion here to make sure you have good type protection:

`const json = await response.json() as User;`